### PR TITLE
CLOUDP-115796: [AtlasCLI] atlas completion is checking for credentials

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-115796

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

**Issue**: Users should be able to run the completion commands without setting credentials.

Failing tests are not related and will be fixed in #1056

### CODE REVIEW

File: internal/cli/root/atlas/builder.go [hunk 1]
```diff
@@ -92,19 +92,7 @@ func Builder(profile *string) *cobra.Command {
...
-			if cmd.Name() == "quickstart" { // quickstart has its own check
-				return nil
-			}
-
-			if strings.HasPrefix(cmd.CommandPath(), fmt.Sprintf("%s %s", atlas, "config")) { // user wants to set credentials
-				return nil
-			}
-
-			if strings.HasPrefix(cmd.CommandPath(), fmt.Sprintf("%s %s", atlas, "auth")) { // user wants to set credentials
+			if skipValidateCredentials(cmd) {
```
Comment (gssbzn) [inline internal/cli/root/atlas/builder.go:95]: [nit] can we flip the question?
```suggestion
			if shouldCheckCredentials(cmd) {
				validate.Credentials()
```
I think it's a bit easier to read

File: e2e/config/autocomplete_test.go [hunk 2]
```diff
@@ -41,3 +41,20 @@ func TestAutocomplete(t *testing.T) {
 		})
 	}
 }
+
+func TestAtlasCLIAutocomplete(t *testing.T) {
```
Comment (gssbzn) [inline e2e/config/autocomplete_test.go:45]: 💯

**Review by gssbzn — DISMISSED** [review]
LGTM

File: internal/cli/root/atlas/builder.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/cli/root/atlas/builder.go]: Updated

File: internal/cli/root/atlas/builder.go [hunk 3]
```diff
@@ -92,19 +92,7 @@ func Builder(profile *string) *cobra.Command {
...
-			if cmd.Name() == "quickstart" { // quickstart has its own check
-				return nil
-			}
-
-			if strings.HasPrefix(cmd.CommandPath(), fmt.Sprintf("%s %s", atlas, "config")) { // user wants to set credentials
-				return nil
-			}
-
-			if strings.HasPrefix(cmd.CommandPath(), fmt.Sprintf("%s %s", atlas, "auth")) { // user wants to set credentials
+			if shouldCheckCredentials(cmd) {
```
Comment (gssbzn) [inline internal/cli/root/atlas/builder.go:95]: it says should but if true you don't check it...

Comment (andreaangiolillo) [reply internal/cli/root/atlas/builder.go:95]: you are right xD, I will update it

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**